tests/components/grid.test.ts

```diff
@@ -1427,6 +1428,34 @@ describe("Copy paste keyboard shortcut", () => {
     expect(getCellContent(model, "A2")).toEqual("things");
   });
 
+  test("cut zone gets cleared on paste if content/style is altered after cut", async () => {
+    setCellContent(model, "A1", "things");
+    setStyle(model, "A1", { bold: true });
+    selectCell(model, "A1");
+    document.body.dispatchEvent(getClipboardEvent("cut", clipboardData));
+    setCellContent(model, "A1", "new content");
+    setStyle(model, "A1", { bold: false });
+    selectCell(model, "A2");
+    document.body.dispatchEvent(getClipboardEvent("paste", clipboardData));
+    expect(getCellContent(model, "A2")).toEqual("things");
+    expect(getStyle(model, "A2")).toEqual({ bold: true });
+    expect(getCell(model, "A1")).toBe(undefined);
+  });
+
+  test("Cut and paste of a date format cell, by toggling formula visibility should return date", async () => {
```

Comment:
it is a broader behaviour than this, I just provided one example. if you look at the code it returns the formattedvalue as long as 1) the cell is not a formula, 2) we don't set 'formulaVisibility' to true. The test should be about 
1) 'cut a non formula cell returns its formatted value'
2) and another test 'cut a formula cell returns the content if `showFormulas === true` where you cut paste with showformulas with true and false values. 
The second test (or equivalent) might already exist though, I have not read the whole file.

Reply:
Pushed the changes :+1: 
Do checkout the previous comment as well.